analyzers/src/SonarAnalyzer.CSharp/Rules/UseParamsForVariableArguments.cs

```diff
@@ -26,37 +26,44 @@ public sealed class UseParamsForVariableArguments : SonarDiagnosticAnalyzer
         internal const string DiagnosticId = "S4061";
         private const string MessageFormat = "Use the 'params' keyword instead of '__arglist'.";
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DescriptorFactory.Create(DiagnosticId, MessageFormat);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterNodeAction(
                 c =>
                 {
-                    var methodDeclaration = (MethodDeclarationSyntax)c.Node;
-
-                    if (methodDeclaration.Identifier.IsMissing ||
-                        methodDeclaration.Modifiers.Any(SyntaxKind.ExternKeyword) ||
-                        !HasAnyArgListParameter(methodDeclaration))
-                    {
-                        return;
-                    }
-
-                    var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodDeclaration);
-                    if (methodSymbol != null &&
-                        methodSymbol.IsPubliclyAccessible() &&
-                        methodSymbol.GetInterfaceMember() == null &&
-                        methodSymbol.OverriddenMethod == null)
+                    if (c.Node.ParameterList() is { Parameters: { Count: > 0 } parameters }
+                        && parameters.Any(p => p.Identifier.IsKind(SyntaxKind.ArgListKeyword))
```

Comment:
I think we could check only the last parameter (instead of going through all with Any).
AKA, and please correct me if I'm wrong, we cannot have:
`void MyMethod(__arglist, int something) { }`